test/unit/will/testCodicilsDamageDate.js

```diff
@@ -0,0 +1,233 @@
+                'codicilsDamageDateKnown': 'optionYes',
+                'codicilsdamagedate-day': '4',
+                'codicilsdamagedate-month': '10',
+                'codicilsdamagedate-year': '3000'
+            };
+            const errorsIn = [];
+            const formdata = {};
+            const session = {};
+            const [, errors] = CodicilsDamageDate.handlePost(ctxIn, errorsIn, formdata, session);
+            expect(errors).to.deep.equal([{
+                'field': 'codicilsdamagedate',
+                'href': '#codicilsdamagedate',
+                'msg': 'The date of when damages or marks appeared on the codicils must be in the past or today'
+            }]);
+            done();
+        });
+
+        it ('should create an error if negative day, month or year entered', (done) => {
+            const ctxIn = {
+                'codicilsDamageDateKnown': 'optionYes',
+                'codicilsdamagedate-day': '-4',
+                'codicilsdamagedate-month': '10',
+                'codicilsdamagedate-year': '2019'
+            };
+            const errorsIn = [];
+            const formdata = {};
+            const session = {};
+            const [, errors] = CodicilsDamageDate.handlePost(ctxIn, errorsIn, formdata, session);
+            expect(errors).to.deep.equal([{
+                'field': 'codicilsdamagedate',
+                'href': '#codicilsdamagedate',
+                'msg': 'Please enter a valid date'
+            }]);
+            done();
+        });
+
+        it ('should create an error when an invalid date has been entered', (done) => {
+            const ctxIn = {
+                'codicilsDamageDateKnown': 'optionYes',
+                'codicilsdamagedate-day': 'a',
+                'codicilsdamagedate-month': 'b',
+                'codicilsdamagedate-year': 'c'
+            };
+            const errorsIn = [];
+            const formdata = {};
+            const session = {};
+            const [, errors] = CodicilsDamageDate.handlePost(ctxIn, errorsIn, formdata, session);
+            expect(errors).to.deep.equal([{
+                'field': 'codicilsdamagedate',
+                'href': '#codicilsdamagedate',
+                'msg': 'Please enter a valid date'
+            }]);
+            done();
+        });
+
+        it ('should add the date as a formatted string to the context', (done) => {
+            const ctxIn = {
+                'codicilsDamageDateKnown': 'optionYes',
+                'codicilsdamagedate-day': '4',
+                'codicilsdamagedate-month': '10',
+                'codicilsdamagedate-year': '2021'
+            };
+            const [ctx] = CodicilsDamageDate.handlePost(ctxIn);
+            const ctxOut = {
+                'codicilsDamageDateKnown': 'optionYes',
+                'codicilsdamagedate-day': '4',
+                'codicilsdamagedate-month': '10',
+                'codicilsdamagedate-year': '2021',
+                'codicilsDamageDate': '04/10/2021',
```

Comment:
once addZero function is removed, 'codicilsDamageDate': '04/10/2021', will need to be changed to 'codicilsDamageDate': '4/10/2021',